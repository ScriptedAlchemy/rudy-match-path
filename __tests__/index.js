import matchPath from '../src'

test('regular match', () => {
  const isMatch = matchPath('/foo/bar', { path: '/foo/bar' })
  expect(isMatch).toBeTruthy()
})

test('exact match', () => {
  const isMatch = matchPath('/foo/bar', { path: '/foo', exact: true })
  expect(isMatch).toBeFalsy()
})

test('strict match', () => {
  const isMatch = matchPath('/foo', { path: '/foo/', strict: true })
  expect(isMatch).toBeFalsy()
})

test('match with params', () => {
  const match = matchPath('/foo/bar/1', { path: '/foo/:bar/:baz' })
  expect(match).toMatchSnapshot() /*? match */
})
