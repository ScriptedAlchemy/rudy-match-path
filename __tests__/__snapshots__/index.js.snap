// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match with params 1`] = `
Object {
  "isExact": true,
  "params": Object {
    "bar": "bar",
    "baz": "1",
  },
  "path": "/foo/:bar/:baz",
  "url": "/foo/bar/1",
}
`;
